src/hypertable.c

```diff
@@ -1334,6 +1333,44 @@ ts_hypertable_create(PG_FUNCTION_ARGS)
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 				 errmsg("invalid time_column_name: cannot be NULL")));
 
+	created = ts_hypertable_create_from_info(table_relid,
+											 flags,
+											 &time_dim_info,
+											 &space_dim_info,
+											 associated_schema_name,
+											 associated_table_prefix,
+											 &chunk_sizing_info);
+
+	hcache = ts_hypertable_cache_pin();
+	ht = ts_hypertable_cache_get_entry(hcache, table_relid);
+	retval = create_hypertable_datum(fcinfo, ht, created);
+	ts_cache_release(hcache);
+
+	PG_RETURN_DATUM(retval);
+}
+
+/* Creates a new hypertable.
+ *
+ * Flags are one of HypertableCreateFlags.
+ * All parameters after tim_dim_info can be NUL
+ * returns 'true' if new hypertable was created, false if 'if_not_exists' and the hypertable already
+ * exists.
+ */
+bool
+ts_hypertable_create_from_info(Oid table_relid, uint32 flags, DimensionInfo *time_dim_info,
```

Comment:
FWIW, since you are refactoring this anyway, I'd generalize this function take an array of DimensionInfo:s instead of locking this down to one time and one space dimension. No reason to bring limitations of the old interface over to this new one...

Reply:
Probably for a future PR though, since already merged :)